woocommerce/woocommerce-android
# Automatic magic link request experiment

This PR adds the automatic magic link request experiment.

**To test:**
1. Start the app and proceed with logging in with a site address
2. Enter the store address, then email address and tap Continue
4. Notice the original password screen is displayed:

![image](https://user-images.githubusercontent.com/1522856/183366326-34b3ccb5-0d72-410d-9de6-5c17226d6513.png)

5. Now comment out the contents of the `FirebaseRemoteConfigRepository::fetchRemoteConfig()` and change the `AutomaticMagicLinkRequestVariant.CONTROL.name` to `AutomaticMagicLinkRequestVariant.AUTOMATIC.name` of `defaultValues` in the same class
6. Build and run the app again
7. Repeat the same login steps as before **(use non-A8c address to receive a magic link)**
8. After entering the email address, notice the following screen is displayed:
 
![image](https://user-images.githubusercontent.com/1522856/183369512-a2c9fdd5-e529-40a9-8d02-1f121061bdde.png)

9. Tap on the `Open mail` button and verify it opens the email client
10. Check that you received the magic link email
